packages/wonder-blocks-modal/src/components/__tests__/modal-launcher.test.js

```diff
@@ -289,45 +294,113 @@ describe("ModalLauncher", () => {
+            </View>,
         );
 
-        const lastButton = wrapper
-            .find("[data-last-focused-button]")
-            .getDOMNode();
-        // force focus
-        lastButton.focus();
+        const lastButton = await screen.findByTestId(
+            "data-last-focused-button",
+        );
 
         // Launch the modal.
-        wrapper.find("button").simulate("click");
-
-        // wait for styles to be applied
-        await wait();
+        userEvent.click(lastButton);
 
         // Act
         savedCloseModal(); // close the modal
-        wrapper.update();
 
         // Assert
-        expect(document.activeElement).toBe(lastButton);
+        await waitFor(() => {
+            expect(lastButton).toHaveFocus();
+        });
+    });
+
+    test("if `closedFocusId` is passed, shift focus to specified element after the modal closes", async () => {
+        // Arrange
+        const ModalLauncherWrapper = () => {
+            const [opened, setOpened] = React.useState(false);
+
+            const handleOpen = () => {
+                console.log("opening modal");
+                setOpened(true);
+            };
+
+            const handleClose = () => {
+                console.log("closing modal");
+                setOpened(false);
+            };
+
+            return (
+                <View style={{gap: 20}}>
+                    <Button>Top of page (should not receive focus)</Button>
+                    <Button id="button-to-focus-on" testId="focused-button">
+                        Focus here after close
+                    </Button>
+                    <Button
+                        testId="launcher-button"
+                        onClick={() => handleOpen()}
+                    >
+                        Open modal
+                    </Button>
+                    <ModalLauncher
+                        onClose={() => handleClose()}
+                        opened={opened}
+                        closedFocusId="button-to-focus-on"
+                        modal={({closeModal}) => (
+                            <OnePaneDialog
+                                title="Triggered from action menu"
+                                content={<View>Hello World</View>}
+                                footer={
+                                    <Button
+                                        testId="modal-close-button"
+                                        onClick={closeModal}
+                                    >
+                                        Close Modal
+                                    </Button>
+                                }
+                            />
+                        )}
+                    />
+                </View>
+            );
+        };
+
+        render(<ModalLauncherWrapper />);
+
+        // Act
+        // Launch modal
+        const launcherButton = await screen.findByTestId("launcher-button");
+        userEvent.click(launcherButton);
```

Comment:
Nit: This section should probably go before act, b/c in this case the action related to the test is closing the modal, so opening the modal is part of `arrange`.